bundles/org.eclipse.ui.ide.application/src/org/eclipse/ui/internal/ide/application/JUnitTestUtil.java

```diff
@@ -0,0 +1,56 @@
+/*******************************************************************************
+ * Copyright (c) 2024 SAP SE.
+ *
+ * This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License 2.0
+ * which accompanies this distribution, and is available at
+ * https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *     Matthias Becker / Sebastian Ratz - initial API and implementation
+ *******************************************************************************/
+package org.eclipse.ui.internal.ide.application;
+
+import java.util.Set;
+import java.util.concurrent.atomic.AtomicBoolean;
+
+import org.eclipse.core.runtime.Platform;
+import org.eclipse.core.runtime.ServiceCaller;
+import org.eclipse.osgi.service.environment.EnvironmentInfo;
+
+public class JUnitTestUtil {
+	private static Boolean cachedIsJunitTestRunning = null;
+
+	public static boolean isJunitTestRunning() {
+		if (cachedIsJunitTestRunning == null) {
+			try {
+				if (Platform.isRunning()) {
+					AtomicBoolean result = new AtomicBoolean();
+					cachedIsJunitTestRunning = ServiceCaller.callOnce(JUnitTestUtil.class, EnvironmentInfo.class, envInfo -> {
+						String application = envInfo.getProperty("eclipse.application"); //$NON-NLS-1$
+						result.set(application != null && Set.of( //
+								// see org.eclipse.pde.internal.launching.IPDEConstants
+								"org.eclipse.pde.junit.runtime.nonuithreadtestapplication", // //$NON-NLS-1$
```

Comment:
@sratz: from the i-build's log 
https://download.eclipse.org/eclipse/downloads/drops4/I20240923-1800/testresults/ep434I-unit-mac64-java17_macosx.cocoa.x86_64_17/org.eclipse.debug.tests.AutomatedSuite.txt
i see that `org.eclipse.test.uitestapplication`  is used.
Is it missing in this list?

Reply:
Yes, I checked for all the test runners we have in platform / PDE / Tycho and these 3 are missing in the list:

- org.eclipse.pde.junit.runtime.coretestapplicationnonmain
- org.eclipse.test.uitestapplication
- org.eclipse.test.coretestapplication